### PR TITLE
YD-656 Fixed buddy not found by ID issue

### PR DESCRIPTION
**Pull request by Bert-R**
It used to fail in this scenario:
* Richard sends a buddy connect request to Bob
* Bob accepts the request
* Bob overwrites his account before Richard gets the chance to process the buddy connect response message
* Richard now gets a failure on everything because the pending buddy connect response message cannot be processed

The fix is to delete any pending buddy connect response messages of buddies that were deleted while the user was offline.

### CODE REVIEW

File: core/src/main/java/nu/yona/server/subscriptions/service/BuddyService.java [hunk 1]
```diff
@@ -355,26 +355,46 @@ void removeBuddyInfoForRemovedUser(User user, Buddy buddy)
...
-					user.getUserAnonymizedId(), buddy.getId(), user.getDevices(), Status.REJECTED,
-					getDropBuddyMessage(DropBuddyReason.USER_ACCOUNT_DELETED, Optional.empty()));
+			prepareCleanupRemovedBuddyInPendingState(user, buddy);
 		}
 		removeBuddy(user.getId(), buddy);
 	}
 
+	private void prepareCleanupRemovedBuddyInPendingState(User user, Buddy buddy)
+	{
+		messageService.deleteUnprocessedMessages(user, m -> isBuddyConnectResponseFromBuddy(m, buddy));
```
Comment (aristotelos) [inline core/src/main/java/nu/yona/server/subscriptions/service/BuddyService.java]: It *might* occur that at the same time, the buddy status is changed to accepted, isn't it? So shouldn't we to be safe remove all messages from the buddy, not only the unprocessed ones?

File: core/src/main/java/nu/yona/server/subscriptions/service/BuddyService.java [hunk 2]
```diff
@@ -355,26 +355,46 @@ void removeBuddyInfoForRemovedUser(User user, Buddy buddy)
...
+		// Send message to "self", as if the requested user declined the buddy request
+		UUID buddyUserAnonymizedId = buddy.getUserAnonymizedId().orElse(null); //
```
Comment (aristotelos) [inline core/src/main/java/nu/yona/server/subscriptions/service/BuddyService.java]: Empty comment `//`

File: core/src/main/resources/application.properties [hunk 3]
```diff
@@ -12,6 +12,7 @@ spring.jpa.show-sql=false
 spring.jpa.properties.hibernate.session.events.log=false
 # logging.level.org.hibernate.SQL=DEBUG
 # logging.level.org.hibernate=DEBUG
+# logging.level.org.hibernate.type.descriptor.sql=TRACE
```
Comment (aristotelos) [inline core/src/main/resources/application.properties]: This should not go to production, should it?

File: core/src/main/java/nu/yona/server/subscriptions/service/BuddyService.java (hunk 1 above)
Comment (Bert-R) [reply core/src/main/java/nu/yona/server/subscriptions/service/BuddyService.java]: No, this cannot occur. The current user would need to switch it to accepted, but that is impossible for a buddy that overwrote their account.

File: core/src/main/resources/application.properties (hunk 3 above)
Comment (Bert-R) [reply core/src/main/resources/application.properties]: It's commented. I left it in for convenience, like the above ones. Every time I needed to Google to find the log category.

File: core/src/main/java/nu/yona/server/subscriptions/service/BuddyService.java (hunk 2 above)
Comment (Bert-R) [reply core/src/main/java/nu/yona/server/subscriptions/service/BuddyService.java]: This was already part of the code that I extracted into a method. Removed it now.

**Pull request merged**